Convert to using mui-datatables instead of material-table

AndreLemay opened the pull request:
Initial draft to address #48 (also includes work for #22 and #24)

duffn commented:
Awesome. I love so many things about this. There are a few rough edges as you said, but this is a great start!

duffn commented on src/components/shared/Table.js:
```diff
@@ -1,42 +1,67 @@
+          download: true,
+          selectableRows: "none",
+          downloadOptions: {
+            filename: `${title.replace(" ", "-")}_download`,
```
This seems to be different than `material-table` as this appears to need the `.csv` extension.

AndreLemay replied:
Indeed it does... You'd think a button who's tooltip says "download as csv" would automatically add that...

duffn commented on src/components/shared/Table.js:
```diff
@@ -1,42 +1,67 @@
+        }
+        columns={columns}
+        data={data}
+        options={{
```
Can we turn off the print option here?

duffn commented on src/components/shared/Table.js:
```diff
@@ -1,42 +1,67 @@
+          downloadOptions: {
+            filename: `${title.replace(" ", "-")}_download`,
+          },
+          fixedHeaderOptions: {
```
Love this!

duffn commented:
This is looking great. I left a few comments.

The mobile usability is my biggest concern, but I think the horizontal scrolling that you mentioned earlier will make `mui-datatables` look more like the current table on mobile? Because I think I like the current setup better.

<img width="386" alt="Screen Shot 2020-02-24 at 9 04 13 AM" src="https://user-images.githubusercontent.com/3457341/75168929-e7d13080-56e4-11ea-918b-c306849b1b9d.png">
<img width="391" alt="Screen Shot 2020-02-24 at 9 04 22 AM" src="https://user-images.githubusercontent.com/3457341/75168928-e7389a00-56e4-11ea-82c2-7c24e61abb3e.png">

AndreLemay commented:
Yeah I haven't really dug too deep into the mobile stuff. This behavior is just the mui-datatables default responsive mode "stacked". 

I think it should just be a matter of changing that to one of the other options and making sure margins and scrolling work on mobile.

I'm also thinking of trying to stick the first column to the left of the page, that way you can still see what row you're looking at.

duffn commented:
Nice!

AndreLemay commented on src/components/Abilities.js:
```diff
@@ -2,14 +2,17 @@ import React from "react"
-      <Link to={`/abilities/${rowData.generatedId}/`}>{rowData.name}</Link>
-    ),
-    defaultSort: "asc",
+    label: "Name",
```
Not sure what you mean. Are you saying you want all the columns that were not groupable in the current table to be excluded from the filter as well?

duffn replied:
Yep! I'm not sure it makes sense to have a filter for things like name or index for example. I'm happy to be convinced otherwise, though.

AndreLemay replied:
I would agree. Name is already handled by the search, and the index isn't really in a filterable format. I'll check the history and update accordingly.

AndreLemay commented:
Ok, I think I've addressed all your comments, and was also able to get the mobile looking decent, and have the first column sticking to the left side.

duffn commented on src/components/shared/Table.js:
```diff
@@ -1,42 +1,105 @@
+import useMediaQuery from "@material-ui/core/useMediaQuery"
+import { createMuiTheme, MuiThemeProvider } from "@material-ui/core/styles"
 
 export default function Table({ title, data, columns, marginTop, grouping }) {
```
This is not going to stop me from merging this, but we're not using `grouping` anymore, so can remove it here and from `StatPage` and `abilities`.

AndreLemay replied:
Yeah, I thought of that, but like I mentioned either here or on the issue, I think it should be possible to do some type of grouping (probably similar to how I did it on my site), so I figured I'd just leave it as a placeholder for now.

duffn replied:
Great, let's leave it then, thanks.

duffn commented:
I've been poking around at this and it looks great! The sticky header and column are going to be big wins.

There are a few nits and questions, but nothing that is going to stop me from merging this today at this point.
